cmd/namespace-check/main.go

```diff
@@ -31,65 +34,47 @@ var (
 		"opa",
 		"velero",
 	}
-)
-
-func init() {
-	checkclient.Debug = true
-}
 
-type Options struct {
-	client kubernetes.Interface
-}
-
-func main() {
-	// create context
-	ctx := context.Background()
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
 
-	// Create a kubernetes client.
-	var err error
-	o := Options{}
-	o.client, err = kubeClient.Create(kubeConfigFile)
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		errorMessage := "failed to create a kubernetes client with error: " + err.Error()
-		reportErr := checkclient.ReportFailure([]string{errorMessage})
-		if reportErr != nil {
-			log.Fatalln("error reporting failure to kuberhealthy:", reportErr.Error())
-		}
-		return
+		log.Fatalln(err.Error())
 	}
-	log.Infoln("Kubernetes client created.")
 
-	ok, err := o.namespaceExist(ctx)
-	if ok {
-		reportErr := checkclient.ReportSuccess()
+	if err := doExpectedNamespacesExist(context.Background(), clientset, namespaces); err != nil {
+		reportErr := checkclient.ReportFailure([]string{"Namespace check failed:" + err.Error()})
 		if reportErr != nil {
-			log.Fatalln("error reporting to kuberhealthy:", err.Error())
-
+			log.Fatalln("Unable to communicate with kuberhealthy", reportErr.Error())
 		}
-		return
+		log.Fatalln("Error checking for namespaces", err)
 	}
 
-	reportErr := checkclient.ReportFailure([]string{"Namespace check failed:" + err.Error()})
+	// report success to Kuberhealthy. If it fails, fail the check.
+	reportErr := checkclient.ReportSuccess()
 	if reportErr != nil {
 		log.Fatalln("error reporting to kuberhealthy:", err.Error())
 	}
 }
 
-func (o Options) namespaceExist(ctx context.Context) (bool, error) {
-	var notFoundNamespaces []string
-	// range over namespaces and check if exists
-	for _, ns := range namespaces {
-		_, err := o.client.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
+// doExpectedNamespacesExist checks if the expected namespaces exist in the cluster.
+func doExpectedNamespacesExist(ctx context.Context, client kubernetes.Interface, expectedNamespaces []string) error {
+	for _, ns := range expectedNamespaces {
+		if checkclient.Debug {
+			log.Println("Checking for namespace", ns)
+		}
+		_, err := client.CoreV1().Namespaces().Get(ctx, ns, metav1.GetOptions{})
 		if apierrors.IsNotFound(err) {
-			notFoundNamespaces = append(notFoundNamespaces, ns)
+			return fmt.Errorf("namespace %s not found", ns)
```

Comment:
This returns when the first expected namespace is not found. should we wait for all the expected namespace lookup to be completed before returning error?